api/nfd/go.mod

```diff
@@ -1,7 +1,6 @@
 module sigs.k8s.io/node-feature-discovery/api/nfd
 
-go 1.22.2
-toolchain go1.23.7
+go 1.24
```

Comment:
Is this necessary bump?

Reply:
Globally on my machine I have go1.24 installed. That's also the required version by the main go.mod of the project. I could not generate changes I made to NFD API because it complained about outdated toolchain and go version. After running `go mod tidy` the go.mod from NFD API changed to:

```
go 1.23
toolchain go1.24
```

Probably, that's the better idea just to bump to go 1.23 to enforce smaller change to people.
Any thoughts?

Reply:
Makes perfect sense.